src/ledger/test/LedgerCloseMetaStreamTests.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "crypto/Hex.h"
+#include "crypto/Random.h"
+#include "history/HistoryArchiveManager.h"
+#include "history/test/HistoryTestsUtils.h"
+#include "lib/catch.hpp"
+#include "main/Application.h"
+#include "main/ApplicationUtils.h"
+#include "test/TestUtils.h"
+#include "test/test.h"
+#include "xdr/Stellar-ledger.h"
+
+using namespace stellar;
+
+#ifndef _WIN32
+TEST_CASE("LedgerCloseMetaStream file descriptor - LIVE_NODE",
+          "[ledgerclosemetastreamlive]")
+{
+    // Step 1: open a writable file descriptor.
+    TmpDirManager tdm(std::string("streamtmp-") + binToHex(randomBytes(8)));
+    TmpDir td = tdm.tmpDir("streams");
+    std::string path = td.getName() + "/stream.xdr";
+    int fd = ::open(path.c_str(), O_CREAT | O_WRONLY, 0644);
+    REQUIRE(fd != -1);
+
+    // Step 2: pass it to an application and close some ledgers,
+    // streaming ledgerCloseMeta to the file descriptor.
+    auto cfg = getTestConfig();
+    cfg.METADATA_OUTPUT_FILE_DESCRIPTOR = fd;
+    VirtualClock clock;
+    auto app = createTestApplication(clock, cfg);
+    app->start();
+    while (app->getLedgerManager().getLastClosedLedgerNum() < 10)
+    {
+        clock.crank(true);
+    }
+    Hash hash = app->getLedgerManager().getLastClosedLedgerHeader().hash;
+    while (!app->getLedgerManager().metadataBufferEmpty())
```

Comment:
you should not need to wait for the meta buffer to be empty (that seems to be missing logic in the app shutdown code: the "wrapper process" would not be able to do this) - regardless, I don't think we should support `LIVE_NODE` in this PR as there are too many unknowns around it

Reply:
Fixed in process of making it synchronous.